File: scrape.py
import requests
from bs4 import BeautifulSoup

# Fetch the HTML content of the page
url = 'https://www.stopstalk.com/contests'
response = requests.get(url)
html_content = response.content

# Parse the HTML using Beautiful Soup
soup = BeautifulSoup(html_content, 'html.parser')

# Find the table containing contest information
table = soup.find('tbody')
if table is not None:
    # Find all the rows in the table
    rows = table.find_all('tr')

    # Loop over each row and extract the required information
    for row in rows:  # Skip the header row
        cols = row.find_all('td')
        name = cols[0].text.strip()
        title = cols[1].find('img')['title']
        start_time = cols[2].text.strip()
        duration = cols[3].text.strip()
        contest_link = cols[4].find('a')['href']
        
        # Print the extracted information
        print(name)
        print(title)
        print(start_time)
        print(duration)
        print(contest_link)
        print();
else: print('Data not found.')

File: scrape.js
const axios = require('axios');
const cheerio = require('cheerio');
const moment = require('moment');

const url = 'https://www.stopstalk.com/contests';

axios.get(url).then((response) => {
  const $ = cheerio.load(response.data);
  const rows = $('tbody').find('tr');

  rows.each((i, row) => {
    const cols = $(row).find('td');
    const name = cols.eq(0).text().trim();
    const title = cols.eq(1).find('img').attr('title');
    const startTime = moment(cols.eq(2).text().trim(), 'YYYY-MM-DD HH:mm').format('YYYY-MM-DD hh:mm A');
    const duration = cols.eq(3).text().trim();
    const contestLink = cols.eq(4).find('a').attr('href');
    const calendarLink = cols.eq(5).find('a').attr('href');
    
    if(name){
    // Print the extracted information
      console.log(name);
      console.log(title);
      console.log(startTime);
      console.log(duration);
      console.log(contestLink);
      console.log(calendarLink);
      console.log();
    }
  });
}).catch((error) => {
  console.log(error);
});
